integration-tests/pkg/types/endpoint.go

```diff
@@ -15,10 +15,10 @@ func (n *EndpointInfo) IsActive() bool {
 }
 
 func SortEndpoints(endpoints []EndpointInfo) {
-	sort.Slice(endpoints, func(i, j int) bool { return endpointComparison(endpoints[i], endpoints[j]) })
+	sort.Slice(endpoints, func(i, j int) bool { return CompareEndpoints(endpoints[i], endpoints[j]) })
 }
 
-func endpointComparison(endpoint1 EndpointInfo, endpoint2 EndpointInfo) bool {
+func CompareEndpoints(endpoint1 EndpointInfo, endpoint2 EndpointInfo) bool {
```

Comment:
Unrelated to the change set, the function handling of nil looks strange: a comparison will always succeed if a second address or an originator process is nil, which might be quite a surprise to the caller.

Reply:
Agreed, that's a bit odd. I'll rework this function a bit

Reply:
In fact I see that my use of this function is incorrect too. I'll fix it.